src/DistributedComputations/distributed_fft_tridiagonal_solver.jl

```diff
@@ -0,0 +1,333 @@
+using CUDA: @allowscalar
+using Oceananigans.Solvers: BatchedTridiagonalSolver
+using Oceananigans.Solvers: stretched_dimensions, stretched_direction
+using Oceananigans.Solvers: Δξᶠ, compute_main_diagonal!
+
+using Oceananigans.Grids: XZRegularRG, 
+                          XYRegularRG,
+                          YZRegularRG
+
+struct DistributedFourierTridiagonalPoissonSolver{G, L, P, B, R, S, β} 
+                          plan :: P              
+                   global_grid :: G
+                    local_grid :: L
+    batched_tridiagonal_solver :: B
+                   source_term :: R
+                       storage :: S
+                        buffer :: β
+end
+
+architecture(solver::DistributedFourierTridiagonalPoissonSolver) =
+    architecture(solver.global_grid)
+
+"""
+    DistributedFourierTridiagonalPoissonSolver(global_grid, local_grid)
+
+Return an FFT-based solver for the Poisson equation evaluated on a `local_grid`
+with one stretched direction
+
+```math
+∇²φ = b
+```
+
+for `Distributed` architectures.
+
+Supported configurations
+========================
+
+In the following, `Nx`, `Ny`, and `Nz` are the number of grid points of the **global** grid, 
+in the `x`, `y`, and `z` directions, while `Rx`, `Ry`, and `Rz` are the number of ranks in the
+`x`, `y`, and `z` directions, respectively. Furthermore, 'pencil' decomposition refers to a domain 
+decomposed in two different directions (i.e., with `Rx != 1` and `Ry != 1`), while 'slab' decomposition 
+refers to a domain decomposed only in one direction, (i.e., with either `Rx == 1` or `Ry == 1`).
+Additionally, `storage` indicates the `TransposableField` used for storing intermediate results;
+see [`TransposableField`](@ref).
+
+1. Three dimensional configurations with 'pencil' decompositions in ``(x, y)`` 
+where `Ny ≥ Rx` and `Ny % Rx = 0`, and `Nz ≥ Ry` and `Nz % Ry = 0`.
+
+2. Two dimensional configurations decomposed in ``x`` where `Ny ≥ Rx` and `Ny % Rx = 0`
+    
+!!! warning "Unsupported decompositions"
+    _Any_ configuration decomposed in ``z`` direction is _not_ supported.
+    Furthermore, any ``(x, y)`` decompositions other than the configurations mentioned above are also _not_ supported.
+    
+Algorithm for pencil decompositions
+============================================
+
+For pencil decompositions (useful for three-dimensional problems),
+there are two forward transforms, two backward transforms, one tri-diagonal matrix inversion
+and a variable number of transpositions that require MPI communication, dependent on the 
+strethed direction:
+
+- a stretching in the x-direction requires four transpositions
+- a stretching in the y-direction requires six transpositions
+- a stretching in the z-direction requires eight transpositions
+
+!!! warning "Computational cost"
```

Comment:
```suggestion
!!! note "Computational cost"
```

a warning is like a picture of a diseased lung on tobacco: it could make sense, if it might kill you. Expensive simulations are not such a case.